src/AspNetCore/Src/precomp.hxx

```diff
@@ -7,17 +7,39 @@
 //
 // System related headers
 //
-#define NTDDI_VERSION 0x06020000
-#define _WIN32_WINNT 0x0602
 #define _WINSOCKAPI_
+
+#define NTDDI_VERSION 0x06010000
+#define WINVER 0x0601
+#define _WIN32_WINNT 0x0601
+
 #include <windows.h>
 #include <atlbase.h>
 #include <pdh.h>
 
 //#include <ntassert.h>
 #include <Shlobj.h>
 #include <httpserv.h>
+
+// So this "hack" should remove our issue of compiling for win7 without header files.
+// We will force ourselves to believe we are on win8 for the iiswebsocket.h file
```

Comment:
I would rephrase slightly as "We force the Windows 8 version check logic in iiswebsocket.h to succeed even though we're compiling for Windows 7. Then set the version defines back to Windows 7 for the rest of the compilation"

NBD, but because of the magic here we want a nice clear comment :).